### PR TITLE
Add clip function.

### PR DESCRIPTION
**Pull request by takagi**
To begin with, place a WIP status pull request and write some description to make the function's specification clear.

### CODE REVIEW

**Comment by takagi** [comment]
Would you test the GPU code because my current CUDA environment has
the same problem with #780 and another machine is now in order which
needs days before its arrival?

**Comment by takagi** [comment]
My new CUDA machine arrived so I tested the GPU code with it, which passed with a module name fix.

File: tests/chainer_tests/functions_tests/math_tests/test_clip.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+from chainer.testing import attr
+
+
+class TestClip(unittest.TestCase):
+
+    def setUp(self):
+        self.x = numpy.random.uniform(-1, 1, (3, 2)).astype(numpy.float32)
+        # Avoid values around x_min and x_max for stability of numerical
+        # gradient
+        for i in range(self.x.size):
```
Comment (okuta) [inline tests/chainer_tests/functions_tests/math_tests/test_clip.py]: How about using `numpy.ndindex`?

File: chainer/functions/math/clip.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 1)
+        x_type, = in_types
+        type_check.expect(x_type.dtype == numpy.float32)
+
+    def forward_cpu(self, x):
+        return utils.force_array(
+            numpy.clip(x[0], self.x_min, self.x_max)
+        ).astype(numpy.float32),
```
Comment (okuta) [inline chainer/functions/math/clip.py]: Is `astype` needed?

File: chainer/functions/math/clip.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    def backward_cpu(self, x, gy):
+        return utils.force_array(
+            gy[0] * (self.x_min < x[0]) * (x[0] < self.x_max)
+        ).astype(numpy.float32),
```
Comment (okuta) [inline chainer/functions/math/clip.py]: Is `astype` needed?

**Comment by okuta** [comment]
LGTM!

**Comment by okuta** [comment]
Please rebase or merge the master branch.

**Comment by takagi** [comment]
Did it.

**Comment by okuta** [comment]
Thank you!

**Pull request merged**